src/test/regress/upgrade/upgrade_common.py

```diff
@@ -16,89 +14,197 @@ def initialize_temp_dir(temp_dir):
     # Give full access to TEMP_DIR so that postgres user can use it.
     os.chmod(temp_dir, 0o777)
 
+
 def initialize_temp_dir_if_not_exists(temp_dir):
     if os.path.exists(temp_dir):
         return
     os.mkdir(temp_dir)
     # Give full access to TEMP_DIR so that postgres user can use it.
     os.chmod(temp_dir, 0o777)
 
-def initialize_db_for_cluster(pg_path, rel_data_path, settings):
-    subprocess.run(['mkdir', rel_data_path], check=True)
-    for node_name in NODE_NAMES:
+
+def initialize_db_for_cluster(pg_path, rel_data_path, settings, node_names):
+    subprocess.run(["mkdir", rel_data_path], check=True)
+    for node_name in node_names:
         abs_data_path = os.path.abspath(os.path.join(rel_data_path, node_name))
         command = [
-            os.path.join(pg_path, 'initdb'),
-            '--pgdata', abs_data_path,
-            '--username', USER
+            os.path.join(pg_path, "initdb"),
+            "--pgdata",
+            abs_data_path,
+            "--username",
+            USER,
         ]
         subprocess.run(command, check=True)
         add_settings(abs_data_path, settings)
 
 
 def add_settings(abs_data_path, settings):
-    conf_path = os.path.join(abs_data_path, 'postgresql.conf')
-    with open(conf_path, 'a') as conf_file:
+    conf_path = os.path.join(abs_data_path, "postgresql.conf")
+    with open(conf_path, "a") as conf_file:
         for setting_key, setting_val in settings.items():
-            setting = "{setting_key} = \'{setting_val}\'\n".format(
-                setting_key=setting_key,
-                setting_val=setting_val)
+            setting = "{setting_key} = '{setting_val}'\n".format(
+                setting_key=setting_key, setting_val=setting_val
+            )
             conf_file.write(setting)
 
 
-def start_databases(pg_path, rel_data_path):
-    for node_name in NODE_NAMES:
+def create_role(pg_path, port, node_ports, user_name):
+    for port in node_ports:
+        command = "SELECT worker_create_or_alter_role('{}', 'CREATE ROLE {} WITH LOGIN CREATEROLE CREATEDB;', NULL)".format(
```

Comment:
Can we avoid the notices somehow?
```
NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
```

Reply:
After setting the `client_min_messages` to `WARNING`, this should be gone.